rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -2801,12 +2842,46 @@ void game_list_frame::PopulateGameGrid(int maxCols, const QSize& image_size, con
 /**
 * Returns false if the game should be hidden because it doesn't match search term in toolbar.
 */
-bool game_list_frame::SearchMatchesApp(const QString& name, const QString& serial) const
+bool game_list_frame::SearchMatchesApp(const QString& name, const QString& serial, bool fallback) const
 {
 	if (!m_search_text.isEmpty())
 	{
-		const QString search_text = m_search_text.toLower();
-		return m_titles.value(serial, name).toLower().contains(search_text) || serial.toLower().contains(search_text);
+		QString search_text = m_search_text.toLower();
+		QString title_name = m_titles.value(serial, name).toLower();
+
+		// Ignore trademarks when no search results have been yielded by unmodified search
+		static const std::initializer_list<QString> s_ignored_on_fallback = { ":", "-", reinterpret_cast<const char*>(u8"®"), reinterpret_cast<const char*>(u8"©"), reinterpret_cast<const char*>(u8"™") };
+
+		if (fallback)
+		{
+			search_text = search_text.simplified();
+			title_name = title_name.simplified();
+
+			QString title_name_replaced_trademarks_with_spaces = title_name;
+			QString title_name_simplified = title_name;
+
+			for (auto& ignored : s_ignored_on_fallback)
+			{
+				search_text.remove(ignored);
+				title_name.remove(ignored);
+				title_name_replaced_trademarks_with_spaces.replace(ignored, " ");
+			}
```

Comment:
can't you replace this loop with the QRegularExpression overload of QString::replace ?

Reply:
Tried ":.-.®.©.™.", seems not working.